docs/en/week08/08-3.md

```diff
@@ -0,0 +1,198 @@
+title: Generative Models - Autoencoders
+lecturer: Alfredo Canziani
+authors: Nandhitha Raghuram, Xinyi Zhao
+date: 25 March 2021
+---
+
+### Recap: Predictive Latent Variable Model
+As we know latent variables are useful in providing additional information that help in predicting multiple values of the target $y$, assiociated with the input $x$. We have seen earlier, in order to predict the target variable $y$ from latent variable $z$, the energy function is given as,
+$$E(y,z) = [y_1 - g_1(z)]^2 + [y_2 - g_2(z)]^2, y \in Y$$
+where $g = [g_{1}, g_{2}]^{T}$, $\mathbb{R} \rightarrow \mathbb{R}^{2}$ and $g(z) = [w_{1}cos(z), w_{2}sin(z)]$.
+
+In this case, the the latent variable $z$ is one dimensional. However, if the $z$ has a higher dimension, this could lead to an energy function of $0$  everywhere. Hence, in order to ensure lower energy values for more compatible predictions we introduce a regularized factor. 
+
+**Training recap**
+
+Given an observation $y$, training the regualarized latent variable model
+$$
+E(y,z) = C(y,\tilde{y}) + R(z)
+$$ 
+where $\tilde{y} = Dec(z)$,
+
+- compute $F_{\beta}(y) = \text{softmin}_{\beta}[E(y,z)]$
+- Minimise $\mathcal{L}(F_{\beta}(\cdot), Y)$
+
+At zero temperature limit, 
+- Compute $\check{z} = \text{argmin } E(y,z)$, $F_{\infty}(y) = \min{E(y,\check{z})}$
+- Minimise $\mathcal{L}(F_{\infty}(\cdot), Y)$
+
+**Target propagation**
+
+Given an observation $y$, we first compute $\tilde{z}$ which is the initial guess of the latent variable $z$
+$$\tilde{z} = Enc(y)$$
+We then initialize $z$ to $\tilde{z}$, here we must ensure that these values are not too far apart from each other. 
+$$\check{z} = argmin E(y,z)\\
+\text{where}, E(y,z) = C(y, \tilde{y}) + R(z) + D(z,\tilde{z})$$
+We now minimise the loss functional in two steps, minimise $L(F_{\infty}(\cdot), Y)$:
+$$\theta_{Dec} = \theta_{Dec} - \triangledown_{\theta_{Dec}}C(y,\tilde{y})\\ \\
+\theta_{Enc} = \theta_{Enc} - \triangledown_{\theta_{Enc}}D(\check{z},\tilde{z})
+$$
+In this way $z$ is constrained in taking only a set of values.
+
+### Autoencoder (AE)
+<center>
+<img src="{{site.baseurl}}/images/week08/08-3/AE.png" /><br>
+<b>Fig. 1</b>: Autoencoder
+</center>
+
+As shown in *Fig. 1*, to summarize at a high level, a very simple form of AE is as follows:
+
+- First, the autoencoder takes in an input and maps it to a hidden state through an affine transformation $\boldsymbol{h} = f(\boldsymbol{W}_h \boldsymbol{x} + \boldsymbol{b}_h)$, where $f$ is an (element-wise) activation function. This is the **encoder** stage. Note that $\boldsymbol{h}$ is also called the **code**.
+- Next, $\hat{\boldsymbol{x}} = g(\boldsymbol{W}_x \boldsymbol{h} + \boldsymbol{b}_x)$, where $g$ is an activation function. This is the **decoder** stage.
+
+**Reconstruction costs**
+
+When the input is real-valued we use mean squared error loss, 
+$$
+C(y,\tilde{y}) = \mid \mid{y} - \tilde{y} \mid \mid^{2} =  \mid\mid{y - \text{Dec}[\text{Enc}(\tilde{y})]}\mid\mid^{2}
```

Comment:
The norm operator is `\Vert`.
`y` needs `\vect{}`.

Reply:
fixed